lib/sorcery/model.rb

```diff
@@ -142,6 +142,7 @@ def authentication_response(options = {})
       def set_encryption_attributes
         @sorcery_config.encryption_provider.stretches = @sorcery_config.stretches if @sorcery_config.encryption_provider.respond_to?(:stretches) && @sorcery_config.stretches
         @sorcery_config.encryption_provider.join_token = @sorcery_config.salt_join_token if @sorcery_config.encryption_provider.respond_to?(:join_token) && @sorcery_config.salt_join_token
+        @sorcery_config.encryption_provider.pepper = @sorcery_config.pepper if @sorcery_config.encryption_provider.respond_to?(:pepper) && @sorcery_config.pepper
```

Comment:
Should the conditional check `sorcery_config.pepper.present?` here? I believe ruby treats an empty string as a truthy object.

Reply:
Adding an empty `'pepper'` does nothing actually - when appending an empty string at the end of the token, the original token remains as it is. (shall I add test cases for that maybe?)

Also providing empty string explicitly has an effect to [remove the prior pepper](https://github.com/Sorcery/sorcery/blob/5746cec165998fd377769515d7e8ea4cc4539938/spec/shared_examples/user_shared_examples.rb#L522) (usually should not be changed, but would be an expected behavior as we see with `salt_join_token` ). If we ignore empty string, there is no way to remove the pepper once set.

I could add an check here though, if you think it makes better senses.
In that case, the only case we need to clarify is how to treat an blank string (i.e. string with spaces)

```ruby
pry(main)> puts 'true' if "\s"
(pry):101: warning: string literal in condition
true
=> nil
pry(main)> puts 'true' if "\s".present?
=> nil
```

So if we use `present?`, with space-filled tokens, it adds no peppers, while with the original approach, it actually appends spaces at the end of the string (they ends up with different thing as digest)

```ruby
pry(main)> x = BCrypt::Password.create("password\s")
=> "$2a$10$Q4ONgGV.uwtJ1iQX/Sr35u4wlDLZ0A28CKCG2s3nbRYHR1tRxg8rS"
pry(main)> x == "password"
=> false
```

Although it looks like a corner case, I thought it might be better to align this behavior with the original `salt_join_token`.

Having said that, of course, I would be happy to follow your call. What would you think?

Reply:
If there's tests that ensure it doesn't break backwards compatibility, I think that would be ideal. Beyond that, I'll have to dig into this a little deeper to get a better understanding.

Reply:
Just added a couple of test cases to describe how this new feature behaves, especially to double check that no breakage occurs with the update.
Thanks for your input, will wait for your feedback/questions if any.